client/pages/discover-page.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Redirect from '../components/redirect';
+import SearchResult from '../components/search-result';
+import AppContext from '../lib/app-context';
+
+export default class Discover extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchInput: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleFilterClick = this.handleFilterClick.bind(this);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { params } = this.context.route;
+    const endpoint = params.get('filterType');
+    const prevEndpoint = prevProps.route.params.get('filterType');
+    const q = this.state.searchInput;
+    if (!q) {
+      // eslint-disable-next-line no-useless-return
+      return;
+    } else if (prevState.searchInput !== q || prevEndpoint !== endpoint) {
+      fetch(`/api/search/${endpoint}?q=${q}`)
+        .then(res => res.json())
+        .then(result => {
+          this.setState({ result });
+        });
+    }
+  }
```

Comment:
Here we need to make sure we aren't spamming the server. We should _only_ do a fetch if the new/old endpoints are different or the new/old q's are different